### PR TITLE
Bool

### PR DESCRIPTION
**Pull request by Dimanaux**

### CODE REVIEW

File: src/main/java/oo/lang/bool/TrueClass.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package oo.lang.bool;
+
+public class TrueClass extends Bool {
+    public static final TrueClass True = new TrueClass();
+
```
Comment (Dimanaux) [inline src/main/java/oo/lang/bool/TrueClass.java]: We can create as many `True` values as we want. But it supposed to be a singleton class.

File: src/main/java/oo/lang/bool/FalseClass.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package oo.lang.bool;
+
+public class FalseClass extends Bool {
+    public static final FalseClass False = new FalseClass();
+
```
Comment (Dimanaux) [inline src/main/java/oo/lang/bool/FalseClass.java]: We can create as many `False` values as we want. But it supposed to be a singleton class. Forbid all `Bool` instances to be manually created.

File: src/main/java/oo/lang/bool/Bool.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package oo.lang.bool;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class Bool {
```
Comment (Dimanaux) [inline src/main/java/oo/lang/bool/Bool.java]: Cover `oo.lang.bool` with tests.

**Review by Dimanaux — COMMENTED** [review]
Consider all the notes below before merge.

File: src/main/java/oo/lang/bool/FalseClass.java (hunk 2 above)
Comment (kehtolaulu) [reply src/main/java/oo/lang/bool/FalseClass.java]: it IS supposed

File: src/main/java/oo/lang/bool/TrueClass.java (hunk 1 above)
Comment (kehtolaulu) [reply src/main/java/oo/lang/bool/TrueClass.java]: same

**Pull request merged**